src/libpgmoneta/wal.c

```diff
@@ -171,6 +171,10 @@ pgmoneta_wal(int srv, char** argv)
       memset(xlogpos, 0, xlogpos_size);
       memcpy(xlogpos, pgmoneta_query_response_get_data(identify_system_response, 2), xlogpos_size);
 
+      // Assign the current xlogpos to LSN
+      memset(config->servers[srv].current_wal_lsn, 0, MISC_LENGTH);
+      snprintf(config->servers[srv].current_wal_lsn, MISC_LENGTH, "%s", xlogpos);
```

Comment:
Hey I took some closer look into it, and you shouldn't use this xlogpos here. We got that from IDENTIFY_SYSTEM command. So that's the current lsn on the server side, not on our side.

Reply:
A better place to update xlogpos on streaming starts would be when [sending the START_REPLICATION message](https://github.com/pgmoneta/pgmoneta/blob/1be259e66d61ef58cd828894e0e15ffc9723c42e/src/libpgmoneta/wal.c#L192).

Reply:
I see. I'll remove this portion of the code.

Reply:
> A better place to update xlogpos on streaming starts would be when [sending the START_REPLICATION message](https://github.com/pgmoneta/pgmoneta/blob/1be259e66d61ef58cd828894e0e15ffc9723c42e/src/libpgmoneta/wal.c#L192). The naming is a little inconsistent, sorry about that.

Yeah also don't forget about this :)